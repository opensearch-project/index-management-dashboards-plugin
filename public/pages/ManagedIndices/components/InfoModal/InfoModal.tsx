/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import {
  EuiSmallButton,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
  EuiCodeBlock,
  EuiText,
} from "@elastic/eui";

interface InfoModalProps {
  info: object;
  onClose: () => void;
}

const InfoModal = ({ info, onClose }: InfoModalProps) => (
  <EuiOverlayMask>
    {/*
      // @ts-ignore */}
    <EuiModal onCancel={onClose} onClose={onClose} maxWidth={1000}>
      <EuiModalHeader>
        <EuiModalHeaderTitle>
          <EuiText size="s">
            <h2>Managed Index Info</h2>
          </EuiText>
        </EuiModalHeaderTitle>
      </EuiModalHeader>

      <EuiModalBody>
        <EuiCodeBlock language="json" fontSize="s">
          {JSON.stringify(info, null, 4)}
        </EuiCodeBlock>
      </EuiModalBody>

      <EuiModalFooter>
        <EuiSmallButton fill onClick={onClose} data-test-subj="infoModalCloseButton">
          Close
        </EuiSmallButton>
      </EuiModalFooter>
    </EuiModal>
  </EuiOverlayMask>
);

export default InfoModal;
