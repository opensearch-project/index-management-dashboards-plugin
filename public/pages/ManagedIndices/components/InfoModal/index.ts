/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import InfoModal from "./InfoModal";

export default InfoModal;
