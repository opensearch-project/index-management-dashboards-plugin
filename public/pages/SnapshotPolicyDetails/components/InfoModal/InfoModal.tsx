/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import {
  EuiButton,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
  EuiCodeBlock,
} from "@elastic/eui";

interface InfoModalProps {
  info: object;
  onClose: () => void;
}

const InfoModal = ({ info, onClose }: InfoModalProps) => (
  <EuiOverlayMask>
    <EuiModal onClose={onClose} maxWidth={1000}>
      <EuiModalHeader>
        <EuiModalHeaderTitle> </EuiModalHeaderTitle>
      </EuiModalHeader>

      <EuiModalBody>
        <EuiCodeBlock language="json" fontSize="m">
          {JSON.stringify(info, null, 4)}
        </EuiCodeBlock>
      </EuiModalBody>

      <EuiModalFooter>
        <EuiButton fill onClick={onClose} data-test-subj="infoModalCloseButton">
          Close
        </EuiButton>
      </EuiModalFooter>
    </EuiModal>
  </EuiOverlayMask>
);

export default InfoModal;
