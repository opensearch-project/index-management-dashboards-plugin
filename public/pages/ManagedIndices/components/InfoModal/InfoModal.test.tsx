/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent } from "@testing-library/react";
import InfoModal from "./InfoModal";

describe("<InfoModal /> spec", () => {
  it("renders the component", () => {
    render(<InfoModal info={{ message: "some info" }} onClose={() => {}} />);
    // EuiOverlayMask appends an element to the body so we should have two, an empty div from react-test-library
    // and our EuiOverlayMask element
    expect(document.body.children).toHaveLength(2);
    expect(document.body.children[1]).toMatchSnapshot();
  });

  it("calls close when close button clicked", () => {
    const onClose = jest.fn();
    const { getByTestId } = render(<InfoModal info={{ message: "some info" }} onClose={onClose} />);

    fireEvent.click(getByTestId("infoModalCloseButton"));
    expect(onClose).toHaveBeenCalled();
  });
});
