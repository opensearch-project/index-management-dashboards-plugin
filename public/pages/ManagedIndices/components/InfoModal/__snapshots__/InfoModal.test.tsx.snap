// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoModal /> spec renders the component 1`] = `
<div
  class="euiOverlayMask euiOverlayMask--aboveHeader"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="euiModal"
      style="max-width: 1000px;"
      tabindex="0"
    >
      <button
        aria-label="Closes this modal window"
        class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
        type="button"
      >
        EuiIconMock
      </button>
      <div
        class="euiModal__flex"
      >
        <div
          class="euiModalHeader"
        >
          <div
            class="euiModalHeader__title"
          >
            <div
              class="euiText euiText--small"
            >
              <h2>
                Managed Index Info
              </h2>
            </div>
          </div>
        </div>
        <div
          class="euiModalBody"
        >
          <div
            class="euiModalBody__overflow"
          >
            <div
              class="euiCodeBlock euiCodeBlock--fontSmall euiCodeBlock--paddingLarge prismjs language-json"
            >
              <pre
                class="euiCodeBlock__pre euiCodeBlock__pre--whiteSpacePreWrap"
                tabindex="-1"
              >
                <code
                  class="euiCodeBlock__code json"
                >
                  <span
                    class="euiCodeBlock__line"
                  >
                    <span
                      class="token punctuation"
                    >
                      {
                    </span>
                    

                  </span>
                  <span
                    class="euiCodeBlock__line"
                  >
                        
                    <span
                      class="token property"
                    >
                      "message"
                    </span>
                    <span
                      class="token operator"
                    >
                      :
                    </span>
                     
                    <span
                      class="token string"
                    >
                      "some info"
                    </span>
                    

                  </span>
                  <span
                    class="euiCodeBlock__line"
                  >
                    
                    <span
                      class="token punctuation"
                    >
                      }
                    </span>
                  </span>
                </code>
              </pre>
            </div>
          </div>
        </div>
        <div
          class="euiModalFooter"
        >
          <button
            class="euiButton euiButton--primary euiButton--small euiButton--fill"
            data-test-subj="infoModalCloseButton"
            type="button"
          >
            <span
              class="euiButtonContent euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                Close
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;
